src/FlipTest.java

```diff
@@ -0,0 +1,34 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class FlipTest {
+    private UnoFlipModel unoFlipModel;
+    private Card testCard;
+
+    @Before
+    public void setUp(){
+        unoFlipModel = new UnoFlipModel();
+        unoFlipModel.getPlayers().add(new Player("test1"));
+        unoFlipModel.getPlayers().add(new Player("test2"));
+        unoFlipModel.setNumPlayers(2);
+        testCard = new Card(Card.Rank.FLIP, Card.Colour.RED,Card.Rank.FLIP,Card.Colour.ORANGE);
+    }
+
+    @Test
+    public void testExecuteSequence(){
+        unoFlipModel.getTurnSeqs().get(15).executeSequence(testCard); // should be index 18 according to enum
+        assertTrue(unoFlipModel.getTurnDirection());
+        assertEquals(testCard, unoFlipModel.getTopCard()); // make sure the testCard is on top
+        Card.flipSide();
+        assertEquals(Card.Colour.ORANGE, unoFlipModel.getCurrentColour()); // check to see if card was flipped
+        assertEquals(Card.Rank.FLIP, unoFlipModel.getCurrentRank());
```

Comment:
Can we also check the player hand cards and maybe the deck to see if things are flipped there as well? we dont' have to check the entire deck, but check a few of the cards from there

Reply:
its hard to check the cards in a player's hand since its random every time and I can't add a specific card to their hand. Same goes with deck, can't add a card to deck to check if it was flipped or not. I manually tested it and it worked by printing out the colour and rank.